Docs/notification-service.md

```diff
@@ -0,0 +1,169 @@
+# Notification Service and Push
+The [Notification Service](https://developer.apple.com/documentation/usernotifications/modifying_content_in_newly_delivered_notifications) extension is a an app extension that allows Firefox iOS to intercept encrypted payloads from [Autopush](https://autopush.readthedocs.io/en/latest/), decrypting them, and possibly modifying them before displaying them to the user.
+
+This document will cover how the Notification Service is setup, its interaction with both the Autopush and Firefox Accounts system to deliver send tabs.
+
+The document will also cover how it's possible to extend the notification service and push to support other, non Firefox Accounts related push notifications.
+
+
+## Requirements
+To understand why the Notification Service and push work the way they do today, it's important to understand some of the constraits around how Firefox Accounts delivers push notifications.
+
+Firefox Accounts leverages the [Autopush](https://autopush.readthedocs.io/en/latest/) infrastructure to deliver push notifications. Because it uses Autopush, **Firefox Accounts delivers encrypted payloads to the user, that need to be decrypted before being displayed.**
+
+## How a Push Notification is delivered to the User
+
+The above requirement thus requires that the application:
+1. **Register with Autopush**, and recieve a unique URL to pass to Firefox Accounts so Firefox Accounts can send push notifications to the appllications.
+2. **Generate a public/private encryption key pair** to be used to encrypt the communication from Firefox Account to the application.
+3. **Deliver the public encryption key to Firefox Accounts.**
+4. **Persist the private encryption key.**
+5. When Firefox Accounts sends a push notification to Autopush, it will encrypt the message using the public key it recieved from the application.
```

Comment:
```suggestion
5. When Firefox Accounts sends a push notification to Autopush, it will encrypt the message using the public key it received from the application.
```